spacelift/resource_stack.go

```diff
@@ -76,16 +76,26 @@ func resourceStack() *schema.Resource {
 				Required:    true,
 				ForceNew:    true,
 			},
+			"namespace": &schema.Schema{
```

Comment:
This should only be required for Gitlab.

Reply:
Is there any way to verify that on a schema definition level? As far as I see the value of  `Required` and `Optional` can be `bool` type only. 

I'm going to achieve the goal by setting `Optional: true` for the `namespace` and then verifying that in `resourceStackCreate` and `resourceStackUpdate` functions but if there's better way to do it, please let me know.

Reply:
Unfortunately that's a known limitation of Terraform for now - [you can only validate individual fields](https://godoc.org/github.com/hashicorp/terraform/helper/schema#SchemaValidateFunc). So the "standard" way of doing it would indeed be on `create` and `update` actions, which sucks because you don't get any advance warning from the plan.

I can see two solutions if you want to provide a better user experience.

1) Combine provider and namespace, and make it a single field. Perhaps you can still call it a "namespace" but expect an input formatted like `GITLAB/spacelift`. Pretty ugly, right?

2) Make `gitlab` itself a field. So a stack definition for Gitlab would look like:

```hcl
resource "spacelift_stack" "gitlab_stack" {
  gitlab {
    namespace = "spacelift"
  }
  
  repository = "infra"

  # other fields...
}
```

This would require something like that on the provider end:

```go
"gitlab": &schema.Schema{
	Type:     schema.TypeList,
	Optional: true,
	MaxItems: 1,
	Elem: &schema.Resource{
		Schema: map[string]*schema.Schema{
			"namespace": {
				Type:     schema.TypeString,
				Required: true,
			},
		},
	},
},
```

...and there would be some logic involved in building the right GraphQL query from that input. This is a *much* nicer solution, though it does require a bit more work.